docs/content/querying/dimensionspecs.md

```diff
@@ -201,3 +201,28 @@ A null dimension value can be mapped to a specific value by specifying the empty
 This allows distinguishing between a null dimension and a lookup resulting in a null.
 For example, specifying `{"":"bar","bat":"baz"}` with dimension values `[null, "foo", "bat"]` and replacing missing values with `"oof"` will yield results of `["bar", "oof", "baz"]`.
 Omitting the empty string key will cause the missing value to take over. For example, specifying `{"bat":"baz"}` with dimension values `[null, "foo", "bat"]` and replacing missing values with `"oof"` will yield results of `["oof", "oof", "baz"]`.
+
+### Lookup extraction function (EXPERIMENTAL)
+A lookup extraction function is a simple key-value mapping where the key-value mappings are unique to a particular lookup.
+This can be used for re-naming values in a datasource for certain queries. To use a particular lookup, simply set the appropriate lookup for the dimExtractionFn
+```json
+{
+  "type":"lookup",
+  "lookup":{"type":"namespace","namespace":"some_lookup"},
+  "retainMissingValueWith":"Unknown",
+  "uniqueRenames":false
+}
+```
+
+```json
+{
+  "type":"lookup",
+  "lookup":{"type":"namespace","namespace":"some_lookup"},
+  "retainMissingValue":true,
+  "uniqueRenames":false
+}
+```
```

Comment:
can we also add a full example that includes a query for using this functionality?